Backend/src/test/java/ktsnvt/tim1/controllers/AuthenticationControllerIntegrationTests.java

```diff
@@ -0,0 +1,182 @@
+package ktsnvt.tim1.controllers;
+
+import ktsnvt.tim1.DTOs.LoginDTO;
+import ktsnvt.tim1.DTOs.UserDTO;
+import ktsnvt.tim1.repositories.UserRepository;
+import ktsnvt.tim1.repositories.VerificationTokenRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.junit.Assert.*;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+public class AuthenticationControllerIntegrationTests {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    TestRestTemplate testRestTemplate;
+
+    @Autowired
+    VerificationTokenRepository verificationTokenRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Test
+    void login_wrongEmail_errorMessageReturned(){
+        LoginDTO loginDTO = new LoginDTO("ppetrovic@gmail.com", "KtsNvt1+");
+
+        HttpEntity<LoginDTO> entity = new HttpEntity<>(loginDTO);
+
+        ResponseEntity<String> result = testRestTemplate.exchange(createURLWithPort("/login"),
+                HttpMethod.POST, entity, String.class);
+
+        assertEquals(HttpStatus.BAD_REQUEST, result.getStatusCode());
+        assertEquals("Invalid email or password", result.getBody());
```

Comment:
`assertNotNull(result.getBody())` before this check. Same goes for other test methods